exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java

```diff
@@ -493,11 +477,36 @@ public Builder withTimeService(TimeProvider timeProvider) {
     public TestKit build() {
       checkCorrectServiceNumber(services.size());
       checkCorrectValidatorNumber();
-      return newInstance(services, nodeType, validatorCount, timeProvider);
+      List<TestKitServiceInstances> testKitServiceInstances = mergeServiceSpecs();
+      return newInstance(testKitServiceInstances, nodeType, validatorCount, timeServiceSpec);
+    }
+
+    /**
+     * Turn collection of service instances into a list of
+     * {@linkplain TestKitServiceInstances} objects for native to work with.
+     */
+    private List<TestKitServiceInstances> mergeServiceSpecs() {
+      Set<ServiceArtifactId> serviceArtifactIds = services.keySet();
+      checkArgument(serviceArtifactIds.containsAll(serviceArtifactFilenames.keySet()),
```

Comment:
(1) The condition does not match the error message (must be equal).
Please add tests for both conditions (no artifact, no services for artifact, both at the same time for diff artifacts).

(2) The error message is not very helpful. A symmetric difference, which is needed to communicate
all 'lost' items to cover the last condition will not be easy enough to understand either.
Therefore, just split into two conditions:
```java
      // To be able to use the shorter names (serviceIds vs serviceArtifactIds),
      // extract in a method
      Set<ServiceArtifactId> serviceIds = services.keySet();
      Set<ServiceArtifactId> deployedIds = serviceArtifactFilenames.keySet();
      SetView<ServiceArtifactId> artifactsWithNoServices = Sets.difference(deployedIds, serviceIds);
      // check empty
      SetView<ServiceArtifactId> missingArtifacts = Sets.difference(serviceIds, deployedIds);
      // check empty
```

(3) serviceArtifactFilenames.keySet() extract in a variable `deployedArtifact[Id]s`, otherwise
it looks like it compares files with artifactIds.